src/snowflake/snowpark/session.py

```diff
@@ -191,6 +191,38 @@ class Session:
     A :class:`Session` object is not thread-safe.
     """
 
+    class RuntimeConfig:
```

Comment:
This class now doesn't use a dictionary to store config params but instead check whether the config is a `property` of `Session` or the underlying `SnowflakeConnection`. In this way, we avoid storing an extra copy of values and needing to keep the copies in sync.